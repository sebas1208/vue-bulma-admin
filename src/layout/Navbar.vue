<template>
    <nav class="navbar is-dark is-fixed-top is-spaced" role="navigation" aria-label="main navigation">
        <div class="navbar-brand">
            <a id="burger" @click="toogleMenu()" :class="{ 'is-active': isActive }" role="button" class="navbar-burger burger" aria-label="menu" aria-expanded="false" data-target="navbarBasicExample">
                <span aria-hidden="true"></span>
                <span aria-hidden="true"></span>
                <span aria-hidden="true"></span>
            </a>
        </div>
    </nav>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";

@Component
export default class NavBar extends Vue {
    public isActive = false;

    constructor() {
        super();
    }

    public toogleMenu() {
        this.isActive = !this.isActive;
        this.$emit("toggleMenu", this.isActive);
    }
}
</script>