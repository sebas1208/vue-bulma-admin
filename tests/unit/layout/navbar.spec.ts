import { shallowMount, Wrapper } from "@vue/test-utils";
import Navbar from "@/layout/Navbar.vue";

describe("NavBar Test", () => {
    let wrapper: Wrapper<Navbar>;

    beforeEach(() => {
        wrapper = shallowMount(Navbar);
    });

    it("Emits an event when burger menu is clicked", () => {
        wrapper.find("#burger").trigger("click");
        expect(wrapper.emitted("toggleMenu")).toBeTruthy();
    });

    it("Change the icon of burger when is clicked", () => {
        const burger = wrapper.find("#burger");
        burger.trigger("click");
        expect(burger.classes("is-active")).toBe(true);

        burger.trigger("click");
        expect(burger.classes("is-active")).toBe(false);
    });

    it("Snapshot Test", () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
