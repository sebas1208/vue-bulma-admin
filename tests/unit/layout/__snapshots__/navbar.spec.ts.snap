// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Test Snapshot Test 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-dark is-fixed-top"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      id="burger"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
</nav>
`;
